storage/database/db_manager.go

```diff
@@ -648,13 +648,17 @@ func (dbm *databaseManager) CreateMigrationDBAndSetStatus(blockNum uint64) error
 
 	// Store the migration status
 	dbm.setStateTrieMigrationStatus(blockNum)
+	dbm.lockInMigration.Unlock()
```

Comment:
It would be better to call `Unlock` with `defer` to avoid unexpected situation in the future.